network/manager.go

```diff
@@ -395,14 +395,25 @@ func (nm *networkManager) CreateEndpoint(cli apipaClient, networkID string, epIn
 	if err != nil {
 		return err
 	}
+	var delEndpointErr error
+	// any error after this point should also clean up the endpoint we created above
+	defer func() {
+		if delEndpointErr != nil {
+			delErr := nw.deleteEndpoint(nm.netlink, nm.plClient, nm.netio, nm.nsClient, nm.iptablesClient, ep.Id)
+			if delErr != nil {
+				logger.Error("Deleting endpoint after failing to save state failed with", zap.Error(delErr))
+			}
+		}
+	}()
 
 	if nm.IsStatelessCNIMode() {
-		return nm.UpdateEndpointState(ep)
+		delEndpointErr = nm.UpdateEndpointState(ep)
+		return delEndpointErr
 	}
 
-	err = nm.save()
-	if err != nil {
-		return err
+	delEndpointErr = nm.save()
```

Comment:
why renaming from err to delEndpointErr

Reply:
Originally, I wanted to make it clear which error would be checked (that any previous assignment of err wouldn't affect the defer, but since we cover the err != nil case right before the defer that will never occur). I'll change it to err.